tools/generate_chain.py

```diff
@@ -0,0 +1,142 @@
+            "prev_hash blob,"
+            "height bigint,"
+            "in_main_chain tinyint,"
+            "block blob)"
+        )
+
+        wallet = bt.get_farmer_wallet_tool()
+        coinbase_puzzlehash = wallet.get_new_puzzlehash()
+
+        blocks = bt.get_consecutive_blocks(
+            3,
+            farmer_reward_puzzle_hash=coinbase_puzzlehash,
+            pool_reward_puzzle_hash=coinbase_puzzlehash,
+            guarantee_transaction_block=True,
+            genesis_timestamp=uint64(1234567890),
+            time_per_block=30,
+        )
+
+        unspent_coins: List[Coin] = []
+
+        for b in blocks:
+            for coin in b.get_included_reward_coins():
+                if coin.puzzle_hash == coinbase_puzzlehash:
+                    unspent_coins.append(coin)
+            db.execute(
+                "INSERT INTO full_blocks VALUES(?, ?, ?, ?, ?)",
+                (
+                    b.header_hash,
+                    b.prev_header_hash,
+                    b.height,
+                    1,  # in_main_chain
+                    zstd.compress(bytes(b)),
+                ),
+            )
+        db.commit()
+
+        # build 2000 transaction blocks
+        with enable_profiler(False):
+            for k in range(2000):
+
+                start_time = time.monotonic()
+
+                print(f"block: {len(blocks)} unspent: {len(unspent_coins)}")
+                new_coins: List[Coin] = []
+                spend_bundles: List[SpendBundle] = []
+                for i in range(1010):
+                    if unspent_coins == []:
+                        break
+                    c = unspent_coins.pop(random.randrange(len(unspent_coins)))
+                    receiver = wallet.get_new_puzzlehash()
+                    bundle = wallet.generate_signed_transaction(uint64(c.amount // 2), receiver, c)
+                    new_coins.extend(bundle.additions())
+                    spend_bundles.append(bundle)
+
+                coinbase_puzzlehash = wallet.get_new_puzzlehash()
+                blocks = bt.get_consecutive_blocks(
+                    1,
+                    blocks,
+                    farmer_reward_puzzle_hash=coinbase_puzzlehash,
+                    pool_reward_puzzle_hash=coinbase_puzzlehash,
+                    guarantee_transaction_block=True,
+                    transaction_data=SpendBundle.aggregate(spend_bundles),
+                    time_per_block=30,
+                )
+
+                b = blocks[-1]
+                for coin in b.get_included_reward_coins():
+                    if coin.puzzle_hash == coinbase_puzzlehash:
+                        unspent_coins.append(coin)
+                unspent_coins.extend(new_coins)
+
+                if b.transactions_info:
+                    fill_rate = b.transactions_info.cost / test_constants.MAX_BLOCK_COST_CLVM
+                else:
+                    fill_rate = 0
+
+                end_time = time.monotonic()
```

Comment:
you might want to use `time.perf_counter` (or `time.process_time`) as it has a much higher tick rate and is meant for benchmarking https://peps.python.org/pep-0418/

Reply:
this is mostly not used for benchmarking, but to let the user estimate how many hours it will take to produce the output chain. This time is in the order of tens of seconds, the clock precision is not critical.